onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -903,7 +903,15 @@ def _infer_Scan(self, node):
         scan_input_axes = get_attribute(node, 'scan_input_axes', [0]*num_scan_inputs)
         num_scan_states = len(node.input) - num_scan_inputs
         scan_input_axes = [handle_negative_axis(ax, self._get_shape_rank(node, i + num_scan_states)) for i, ax in enumerate(scan_input_axes)]
-        for i, si in enumerate(subgraph.input):
+        # We may have cases where the subgraph has optionial inputs that appear in both subgraph's input and initializer,
+        # but not in the node's input. In such cases, the input model might be invalid, but let's not choke on it.
+        # Instead, let's issue a warning, skip the optional inputs, and keep going forward.
+        subgraph_inputs = subgraph.input
+        if len(subgraph_inputs) != len(node.input):
+            print("warning: skip optionial inputs for scan subgraph {} of scan node {}".format(subgraph.name, node.name))
+            subgraph_initializer_names = [i.name for i in subgraph.initializer]
```

Comment:
>subgraph.initializer [](start = 58, length = 20)

If the offending input is from subgraph initializer, it should be safe to remove it from input without warning, as there's no way to override initializer from subgraph. ORT only allows overriding main graph initializer using optional inputs